docker/CMSRucioClient/scripts/updateDDMQuota

```diff
@@ -10,43 +10,76 @@ client = Client()
-            rucio = source["used"]
-        if source["source"] == "unavailable":
-            unavailable = source["used"]
-        if source["source"] == "expired":
-            expired = source["used"]
-
-    # Normalise
-    if static == 0:
-        continue # Skip if static is 0
-
-    # ddm_quota is set proportional to percentage of (dynamic + free) space
-    # Apprantly, python integers do not overflow, https://docs.python.org/3/library/exceptions.html#OverflowError
-
-    locked = rucio - unavailable - expired
-    ddm_quota = int((max(static - locked, 0)/static)*1e4)
-
-    # Override ddm_quota for operational purposes
-    rse_attributes = client.list_rse_attributes(rse)
-    if "override_ddm_quota" in rse_attributes:
-        ddm_quota = rse_attributes["override_ddm_quota"]
-    
-    if not DRY_RUN:
+def calculate_ddm_quotas():
+    for rse in rses:
+        rse_usage = list(client.get_rse_usage(rse))
+
+        static, rucio, unavailable, expired = 0, 0, 0, 0
+        for source in rse_usage:
+            if source["source"] == "static":
+                static = source["used"]
+            if source["source"] == "rucio":
+                rucio = source["used"]
+            if source["source"] == "unavailable":
+                unavailable = source["used"]
+            if source["source"] == "expired":
+                expired = source["used"]
+
+        # Normalise
+        if static == 0:
+            continue  # Skip if static is 0
+
+        # ddm_quota is set proportional to percentage of (dynamic + free) space
+        # Apprantly, python integers do not overflow, https://docs.python.org/3/library/exceptions.html#OverflowError
+
+        locked = rucio - unavailable - expired
+        ddm_quota = int((max(static - locked, 0) / static) * 1e4) ** 2
+
+        # Override ddm_quota for operational purposes
+        rse_attributes = client.list_rse_attributes(rse)
+        if "override_ddm_quota" in rse_attributes:
+            overridden_ddm_quotas[rse] = rse_attributes["override_ddm_quota"]
+            continue
+
+        ddm_quotas[rse] = ddm_quota
+
+
+def normalize_ddm_quotas():
+    weights = [int(value) for value in ddm_quotas.values()]
+    for rse, weight in ddm_quotas.items():
+        ddm_quotas[rse] = int( ((int(weight) - min(weights)) / (max(weights) - min(weights)) ) * MAX_DDM_QUOTA )
+
+
+def set_ddm_quotas():
+    # Set automatically calculated ddm quotas
+    for rse, ddm_quota in ddm_quotas.items():
+        client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
+        print("Set ddm_quota for {} to {}".format(rse, ddm_quota))
+
+    # Set overriden ddm_quotas
+    for rse, ddm_quota in overridden_ddm_quotas.items():
         client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
-    print("Setting ddm_quota for {} to {}".format(rse, ddm_quota))
+        print("Override ddm_quota for {} to {}".format(rse, ddm_quota))
+
+def main():
+
+    calculate_ddm_quotas()
+    normalize_ddm_quotas()
+    if not DRY_RUN:
```

Comment:
The purpose of dry_runsis to print out the changes without actually doing it.
You can pass the variable and check it within the set_ddm_quotas function and only set it when DRY_RUN is false.

Reply:
Indeed, thanks for catching it. I fixed it.